### PR TITLE
No issue: Refactor web page items clicks

### LINKED ISSUE #7856: Intermittent UI test failure - < WebControlsTest. verifyDismissTextSelectionToolbarTest >
Disabled in #7449

### PR DESCRIPTION
**Pull request by AndiAJ**
No issue: Refactor web page items clicks.

❗Observations:
-  Some items only work if it matches exactly with the string in our input argument created `webPageItemWithText` for these cases.
- And in other cases it should only contain the string in our input argument. `webPageItemContainingText`

✅ Successfully ran 25x the affected UI tests on Firebase

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [ ] **QA Needed**

### To download an APK when reviewing a PR:
1. click on `Show All Checks`,
2. click `Details` next to `build-focus-debug` or `build-klar-debug` for changes targeting Klar,
2. click `View task in Taskcluster`,
3. click the `Artifacts` row,
4. click to download any of the apks listed here which use an appropriate name for each CPU architecture.

### GitHub Automation
Fixes #7856

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt [hunk 1]
```diff
@@ -75,13 +76,19 @@ class BrowserRobot {
...
-        mDevice.findObject(UiSelector().textContains("Get Location")).waitForExists(waitingTime)
-        mDevice.findObject(UiSelector().textContains("Get Location")).click()
+        webPageItemContainingText("Get Location")
+            .also {
+                it.waitForExists(waitingTime)
+                it.click()
+            }
     }
 
     fun clickGetCameraButton() {
```
[the remark this answers is not in the document]
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt]: Done !

File: app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt [hunk 2]
```diff
@@ -75,13 +76,19 @@ class BrowserRobot {
     }
 
     fun clickGetLocationButton() {
```
[the remark this answers is not in the document]
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt]: Done !

File: app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt [hunk 3]
```diff
@@ -113,29 +120,7 @@ class BrowserRobot {
...
-                    browserScreen {
-                    }.openMainMenu {
-                    }.clickReloadButton {
-                    }
-                    progressBar.waitUntilGone(waitingTime)
-                }
-            }
-        }
-    }
+    fun longPressLink(linkText: String) = longClickPageObject(webPageItemWithText(linkText))
```
[the remark this answers is not in the document]
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt:111]: I've tried but it looks like for link clicks we'll need to match exactly the text (`UiSelector().text()`), whereas for page text, we'll need to query the text using `textContains()`.

File: app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt [hunk 4]
```diff
@@ -424,8 +304,7 @@ class BrowserRobot {
         var currentTries = 0
```
[the remark this answers is not in the document]
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt]: See [comment](https://github.com/mozilla-mobile/focus-android/pull/7835#discussion_r994512483)

**Review by sv-ohorvath — APPROVED** [review]

**Review by AaronMT — APPROVED** [review]

**Pull request merged**